polkadot-host-spec/c02-state.tm

```diff
@@ -476,42 +476,193 @@
     <math|H<around*|(|R|)>>.
   </definition>
 
+  <subsection|Changes Trie><label|defn-changes-trie>
+
+  Polkadot focuses on light client friendliness and therefore implements
+  functionality which allows identifying changes in the blockchain without
+  requiring to search through the entire chain. The <strong|Changes Trie> is
+  a datastructure maintained by the Polkadot node which tracks storage
+  changes created by each block. The primary method of generating the Changes
+  Trie is by calling the <verbatim|ext_storage_changes_root> Host API as
+  described in section <reference|sect-ext-storage-changes-root>. The
+  Polkadot Host requires a mechanism to identify the changes created by a
+  block, as mentioned in section <reference|sect-state-storage>.
+
   \;
+
+  The Changes Trie tracks changes in three different types of mappings (or
```

Comment:
When we talk about "Changes Trie" I think at some point we need to formally define it as a trie. Is it radix-16 trie for example. Are the nodes have exactly the same structure as the state trie etc. So maybe turning this paragraph into a definition saying what type of tree it is and then the key type and value type etc. 
"Track changes in three different types" what does this algorithmically mean? Like we store the key that is change in a trie node? it is not really clearly at least from the first read.

Reply:
I'll reference the radix-16 trie in the Changes Trie section.

> "Track changes in three different types" what does this algorithmically mean?

Depending on the enum, different value types gets inserted into that trie. That type is identified by the enum index. The mappings are:

```
<changed-key> --> <array-of-extrinsic-indexes>
<changed-key> --> <array-of-block-numbers>
<changed-key> --> <array-of-child-changes-tries>
```

I'll change the "tracks changes *in*..." to "tracks changes *of*" this as:

> The Changes Trie tracks changes of three different types of mappings (or search indices) which are differentiated by their key as defined in table 2.1.